src/features/SignAndVerifyMessage/signMessage.slice.ts

```diff
@@ -0,0 +1,99 @@
+import { getAddress } from '@ethersproject/address';
+import { createAction, createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { all, call, put, takeLatest } from 'redux-saga/effects';
+
+import { IFullWallet } from '@services/WalletService';
+import type { AppState } from '@store';
+import { ISignedMessage, WalletId } from '@types';
+import { addHexPrefix } from '@utils';
+
+interface State {
+  status: 'SIGN_SUCCESS' | 'SIGN_REQUEST' | 'SIGN_FAILURE' | 'INIT';
+  walletId?: WalletId;
+  error?: true;
+  message?: string;
+  signedMessage?: ISignedMessage;
+}
+
+export const initialState: State = {
+  status: 'INIT'
+};
+
+export const signMessageSlice = createSlice({
+  name: 'signMessage',
+  initialState,
+  reducers: {
+    walletSelect(state, action: PayloadAction<WalletId>) {
+      state.walletId = action.payload;
+    },
+    messageUpdate(state, action: PayloadAction<string>) {
+      state.message = action.payload;
+      state.status = 'INIT';
+      delete state.error;
```

Comment:
Does `Immutable.js` handle this?

Reply:
`createSlice` uses `Immer` which accepts `delete`: https://immerjs.github.io/immer/update-patterns/#object-mutations